observer/test/observer.test.js

```diff
@@ -7,8 +7,7 @@ import { observeTransferEvents, observeScheduledRewards } from '../lib/observer.
 
 describe('observer', () => {
   let pgPools
-  const getDayAsISOString = d => d.toISOString().split('T')[0]
-  const today = () => getDayAsISOString(new Date())
+  const today = () => new Date().toLocaleDateString('en-CA')
```

Comment:
The existing way was causing a mismatch between the hours of 8pm - 12am in EST, because it would use UTC, while the SQL now() calls default to the local time zone.

What do you guys think of this change here, and the similar one in request-helpers.js?

Reply:
IIUC, you are changing today from "today in UTC" to "today in Canada". Isn't this going to break the tests for people in Europe/Asia/Australia?

> the SQL now() calls default to the local time zone

That depends on the time zone configured on the machine running the server. I am using Docker and `now()` returns time in UTC - the same value as `new Date().toISOString()` in JavaScript.

Anyhow, I support fixing the way how we build test data to work in all time zones.

Can we make a DB query to give us the current date?

Something along the following lines:

```js
let today

// ...
before(async() => {
  // ...
  today = (await pgPools.stats.query('SELECT CURRENT_DATE AS today')).rows[0].today
})
```

Reply:
I think I don't fully understand what exactly is performed by `new Date().toLocaleDateString('en-CA')`.

Are you using CA instead of e.g. US to get the date formatted in YYYY-MM-DD format? Is it safe to rely on the CA locale using the ISO Date format? Locales can change over time, e.g. Canada switched to ISO Date format relatively recently. (It didn't use it [until at least 2016](https://www.cbc.ca/news/canada/proposed-legislation-aims-to-settle-date-debate-1.3407640).)

Based on [this StackOverflow answer](https://stackoverflow.com/a/78714460/69868) (that's not correct for formatting date-time), I think the following could work for us since we are interested only in the date part:

```js
const today = () => new Date(Date.now() - new Date().getTimezoneOffset() * 60000).toISOString().split('T')[0]
```

Thoughts?

Reply:
This works for me! Will try it out

Reply:
Ah didn't notice your first comment in this thread. I was using it get the date formatted in YYYY-MM-DD, as suggested in that StackOverflow answer and elsewhere, but I do see the issue with potential future locale date format changes.

I personally prefer your second comment's idea as opposed to the DB query, but it's up to you!

Reply:
On the second though, I am fine with either your current implementation or the `getTimezoneOffset()` solution. I'd rather avoid making a DB roundtrip if we have a better option.

What's important is to add a code comment clearly spelling out:
1. What we are trying to achieve - get the current date in the current timezone in YYYY-MM-DD format
2. How the implementation works - if using your solution, explain that Canada uses the ISO date format.